jobs/tests/test_to_kinto.py

```diff
@@ -2,37 +2,87 @@
-        self.cache_file = os.path.join(tempfile.gettempdir(), 'cache.dat')
 
-    def tearDown(self):
-        try:
-            os.path.remove(self.cache_file)
-        except Exception:
-            pass
+    @pytest.fixture(autouse=True)
+    def init_cache_files(self, tmpdir):
+        # Use str() on these LocalPath instances to turn them into plain
+        # strings since to_kinto.fetch_existing() expects it to be a string.
+        self.cache_file = str(tmpdir.join('cache.json'))
+        self.old_cache_file = str(tmpdir.join('old.json'))
 
     def test_records_are_not_duplicated(self):
-        mocked = mock.MagicMock()
+        mocked = unittest.mock.MagicMock()
+
         mocked.session.server_url = 'http://localhost:8888/v1'
         # First, populate the cache.
         mocked.get_records.return_value = [
             {'id': 'a', 'title': 'a', 'last_modified': 1}
         ]
-        fetch_existing(mocked, self.cache_file)
+        first = fetch_existing(mocked, cache_file=self.cache_file)
+        assert isinstance(first, dict)
+        assert len(first) == 1
+        assert first['a'][0] == 1  # [0] is the last_modified
+        first_hash = first['a'][1]  # [1] is the hash string
 
         mocked.get_records.return_value = [
             {'id': 'a', 'title': 'b', 'last_modified': 2}
         ]
-        second = fetch_existing(mocked, self.cache_file)
-
+        second = fetch_existing(mocked, cache_file=self.cache_file)
+        assert isinstance(first, dict)
         assert len(second) == 1
-        assert second[0]['title'] == 'b'
+        assert second['a'][0] == 2
+        second_hash = second['a'][1]
+        assert first_hash != second_hash
+
+    def test_dump_file_cache_migration(self):
+        # Make sure the new cache file doesn't exist
+        if os.path.exists(self.cache_file):
+            os.remove(self.cache_file)
+        # The old dump used to a list of records.
+        with open(self.old_cache_file, 'w') as f:
+            records = [
+                {'id': 'a', 'last_modified': 1, 'title': 'a'}
+            ]
+            json.dump(records, f)
+        mocked = unittest.mock.MagicMock()
+        mocked.session.server_url = 'http://localhost:8888/v1'
+        # First, populate the cache.
+        mocked.get_records.return_value = [
+            {'id': 'a', 'title': 'b', 'last_modified': 2},
+        ]
+        fetch_existing(
+            mocked,
+            cache_file=self.cache_file,
+            old_cache_file=self.old_cache_file
+        )
+        assert not os.path.exists(self.old_cache_file)
+        with open(self.cache_file) as f:
+            records = json.load(f)
+            assert len(records) == 1
+            assert records['a'][0] == 2  # [0] is last modified
+
+        # The migration is done, but let's make sure the fetch_existing
+        # continues to work as expected.
+        mocked.get_records.return_value = [
+            {'id': 'a', 'title': 'b', 'last_modified': 2},
```

Comment:
Since the migration is done, I suppose it should use the latest Etag, and this record does not seem unchanged, so it should not be listed here.

In order to test that the entries present in the old cache effectively end up in the new cache, it would make sense not to list this record here, so that when you make sure it is present at the end, you know it's taken from the old cache file